tests/system/test_diags.py

```diff
@@ -56,11 +56,14 @@ def move_to_web(machine_path_re_str, html_prefix_format_str, results_dir):
 
 def count_images(directory):
     images = []
-    for _, _, filenames in os.walk(directory):
-        for file in filenames:
-            if file.endswith(".png"):
-                images.append(file)
-    return len(images)
+    for root, _, filenames in os.walk(directory):
+        # download_data.py won't download files in the viewer directory
+        # because the webpage is more than a simple page of links.
```

Comment:
The `download_data` script expects a plain HTML page listing links. Once `viewer` is reached, the HTML isn't so plain. The `complete_run` test is able to check images in `viewer` because it looks at the files directly -- it doesn't download them from the web.